hydrolib/core/dflowfm/ini/models.py

```diff
@@ -204,6 +207,34 @@ def convert_value(
 
         return value
 
+    @validator("datablock")
+    def _validate_no_nans_are_present(cls, datablock: Datablock) -> Datablock:
+        """Validates that the datablock does not have any NaN values.
+
+        Args:
+            datablock (Datablock): The datablock to verify.
+
+        Raises:
+            ValueError: When a NaN is present in the datablock.
+
+        Returns:
+            Datablock: The validated datablock.
+        """
+        for list in datablock:
+            for value in list:
+                if cls._is_float_and_nan(value) or cls._is_string_and_nan(value):
+                    raise ValueError("NaN is not supported in datablocks.")
```

Comment:
You could do this shorter:
```python
if any(cls._is_float_and_nan(value) for list in datablock for value in list):
    raise ValueError("NaN is not supported in datablocks.")
```
Not sure whether that's more readable. (but SonarCloud always gives me penalty points for deeply nested control structures)

Reply:
The complexity remains the same, but I agree that your suggestion may be the way most people do it in python, so I will use your suggestion! 👍